### ISSUE TITLE
Enhancement: event_rpcgen.py could leverage CFFI or Swig for parsing/outputting

### ISSUE THREAD

**Issue by ngie-eign** [libevent/libevent#989]
The primary purpose of `event_rpcgen.py` is to create libevent compatible RPC structures for end-users.

While churning over the code in #987, I've been thinking about the overall architecture potentially being simplified via CFFI integration, as CFFI has hooks for producing C/C++ extensions.

Swig is another alternative method that could be used instead; I see a TODO comment that references creating additional language bindings--Swig is very good at that.

**Comment by azat** [comment]
Indeed, great thoughts in general!

I used swig for generating python bindings from C code, and it creates tons of wrappers, but if this can be avoided then I'm fine with swig too.

**Comment by ngie-eign** [comment]
> Indeed, great thoughts in general!
> 
> I used swig for generating python bindings from C code, and it creates tons of wrappers, but if this can be avoided then I'm fine with swig too.

Yeah. That's a good point. This is something, by the way, that apache thrift leverages as well, which I think is akin to what libevent tries to do.
